cli_tools/common/assert/assert.go

```diff
@@ -32,3 +35,13 @@ func Contains(element string, arr []string) {
 	}
 	panic(fmt.Sprintf("%s is not a member of %v", element, arr))
 }
+
+// DirectoryExists asserts that a directory is on the current filesystem.
+func DirectoryExists(dir string) {
+	stat, err := os.Stat(dir)
+	if os.IsNotExist(err) {
+		panic(fmt.Sprintf("%s: Directory not found", dir))
+	} else if !os.IsNotExist(err) && !stat.IsDir() {
+		panic(fmt.Sprintf("%s: Not a directory", dir))
+	}
+}
```

Comment:
Since this doesn't use only input parameters to determine the output, the location doesn't seem appropriate. FileSystemUtils or similar would be a better location, and outside assert package. Otherwise, any function that returns a boolean could end up in this package.

Reply:
A couple of items:
 * In common, all of the functions are precondition assertions that "should" always pass, and if they don't pass, it's a programming or packaging bug.
 * From a reader's point of view, the symbol is: `assert.DirectoryExists`
 * There isn't a return value for this or the other functions

Given this, would you be alright:
1. Keep the package `assert`
2. Move to a different file, eg: `filesystem_assertions.go`

Reply:
`DirectoryExists` seems like a generic function that is currently used as an assertion but could be used to control program flow in the future. I think it would be more discoverable in a package that is not focused on assertions.

Maybe have both, one in file utils package that returns a boolean and one in assert that throws panic based on the output from the first?